jbmc/regression/jbmc/short_arg_to_int_param/ShortToInt.java

```diff
@@ -0,0 +1,10 @@
+public class ShortToInt {
+    public void main(int arg) {
+        intMethod((short)arg);
+        assert true;
+    }
+
+    public int intMethod(int i) {
+        return i;
+    }
+}
```

Comment:
So did this test actually fail before? It's not obvious to me what was causing trouble here before (but also I'm a Java noob). I think I would expect an assertion over the value of `i`.

Reply:
The issue was to do with the `intMethod((short)arg);` call, I just saw this shown on a method that also returned an int so kept that little detail. The invariant here: https://github.com/diffblue/cbmc/blob/9bc196befd95489e789d49429a5dc70d29335bff/src/goto-programs/remove_virtual_functions.cpp#L166 fails, because the `intMethod((short)arg);` argument is a short (due to the cast) but the parameter is expecting an int.

Reply:
(and the JVM spec actually says the types must be correct on the stack upon execution of a call instruction, we just implemented the cast opcodes wrongly before so that wasn't always true)

Reply:
Ok, thanks both for the elaboration. I think it would be good to squash the two commits as otherwise there isn't sufficient context to understand what this test is actually testing.